### ISSUE TITLE
selectByGroup and allowDeselectOption enabled

### ISSUE THREAD

**Issue by mrgrashalm** [brianvoe/slim-select#212]
**Behaviour:**
When there are some of the options in a group selected and you want to select the whole group by cklicking on the groupname, the selection inverts.
The selected get deselected and the other way around.

Normaly I would expect SlimSelect to select all options in the group, even if there are some already selected.

Is there an easy way to get that behaviour?

**Comment by a-mellor** [comment]
@mrgrashalm did you manage to find a way to fix this?

**Comment by ondrop** [comment]
Hi. I fixed it like that. There is my example
https://codesandbox.io/s/slimselect-optgroup-logic-ltogkb?file=/src/index.js

I tried to catch click event on optgroup-label block and write new logic, but because of e.stopPropogation i couldn't.
In this solution I replace optgroup-label div and rewrite logic to optgroup-label block.
Whole options in group will be selected if you click on label. And option would be unselected if they all were selected before
